### PR TITLE
KB - Validate report parameters on the client-side

### PR DESCRIPTION
**Pull request by vegeorgi**

### CODE REVIEW

File: knowledge-base/how-to-validate-report-parameters-on-the-client.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+## Description
+
+Client-side validation on a report parameter based on the value of another report parameter
+
+## Solution
+
+The following guide shows how to validate a report parameter in [HTML5 ASP.NET MVC Report Viewer](../html5-report-viewer). The approach could be applied to any other Web Report Viewer.
+
+1. Start by creating custom parameter editors by defining them through the [ParameterEditors method](https://docs.telerik.com/reporting/m-telerik-reportviewer-mvc-ireportviewerbuilder-parametereditors) when creating the report viewer widget:
```
Comment (silviyaboteva) [inline knowledge-base/how-to-validate-report-parameters-on-the-client.md]: use relative paths to the reporting docs, for example ../m-telerik-reportviewer-mvc-ireportviewerbuilder-parametereditors.

You may edit the rest of the articles' links too.

File: knowledge-base/how-to-validate-report-parameters-on-the-client.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	<tr>
+		<td>Report Viewer</td>
+		<td>Web Viewers</td>
+	</tr>
+</table>
+
+
+## Description
+
+Client-side validation on a report parameter based on the value of another report parameter
```
Comment (atkeranov) [inline knowledge-base/how-to-validate-report-parameters-on-the-client.md]: Could be more descriptive about the problem and its implications.

File: knowledge-base/how-to-validate-report-parameters-on-the-client.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+## Solution
+
+The following guide shows how to validate a report parameter in [HTML5 ASP.NET MVC Report Viewer](../html5-report-viewer). The approach could be applied to any other Web Report Viewer.
```
Comment (atkeranov) [inline knowledge-base/how-to-validate-report-parameters-on-the-client.md]: Web Report viewer is misleading. The old ASP.NET viewer is a web report viewer too.

File: knowledge-base/how-to-validate-report-parameters-on-the-client.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  
+2. In this example, we have a single select editor with available values and a number parameter that has to be validated based on the first parameter value. That means that we are going to need a combobox editor and simple input editor.
+In this case, we have both UI elements that we are needing to represent our editors. The first one will be a [Kendo ComboBox widget](https://docs.telerik.com/kendo-ui/api/javascript/ui/combobox), which we have from the [Telerik Reporting](https://docs.telerik.com/reporting/html5-report-viewer-widget-requirements) scripts and
+simple [HTMl5 text input](https://www.w3schools.com/tags/tag_input.asp). If you wish to use some Kendo widget, that is not included in a subset of the used by default Kendo UI widgets from Telerik Reporting, you need to add manually the scripts for the additional widget or add [kendo.all.js](https://docs.telerik.com/kendo-ui/intro/installation/cdn-service):
+
+3. Create the editors and applay the validation on [input change](https://api.jquery.com/change/):
```
Comment (atkeranov) [inline knowledge-base/how-to-validate-report-parameters-on-the-client.md]: Typo "applay".

File: knowledge-base/how-to-validate-report-parameters-on-the-client.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+                    validatingParameter.trigger("change");
+                }
+            }
+
+            return {
+                beginEdit: function (param) {
+
+                    parameter = param;
+
+                    //Find the parameter based on its name if you have more then one parameter of this type
```
Comment (atkeranov) [inline knowledge-base/how-to-validate-report-parameters-on-the-client.md]: Typo "then".

File: knowledge-base/how-to-validate-report-parameters-on-the-client.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+                    basedParameterValue = val;
+                    //Trigger change event on the validatingParameter to fire the validation
+                    validatingParameter.trigger("change");
+                }
+            }
+
+            return {
+                beginEdit: function (param) {
+
+                    parameter = param;
```
Comment (atkeranov) [inline knowledge-base/how-to-validate-report-parameters-on-the-client.md]: Is this needed?

File: knowledge-base/how-to-validate-report-parameters-on-the-client.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  
+2. In this example, we have a single select editor with available values and a number parameter that has to be validated based on the first parameter value. That means that we are going to need a combobox editor and simple input editor.
+In this case, we have both UI elements that we are needing to represent our editors. The first one will be a [Kendo ComboBox widget](https://docs.telerik.com/kendo-ui/api/javascript/ui/combobox), which we have from the [Telerik Reporting](https://docs.telerik.com/reporting/html5-report-viewer-widget-requirements) scripts and
+simple [HTMl5 text input](https://www.w3schools.com/tags/tag_input.asp). If you wish to use some Kendo widget, that is not included in a subset of the used by default Kendo UI widgets from Telerik Reporting, you need to add manually the scripts for the additional widget or add [kendo.all.js](https://docs.telerik.com/kendo-ui/intro/installation/cdn-service):
+
+3. Create the editors and applay the validation on [input change](https://api.jquery.com/change/):
+
+    ```
+    var basedParameterName = "Parent", //The parameter name that will be used for validaion
```
Comment (atkeranov) [inline knowledge-base/how-to-validate-report-parameters-on-the-client.md]: The naming of "based" and "validating" could be improved. Base: master, parent. Validating: child, dependent.

**Pull request merged**